### PR TITLE
Fix: Re-migrate Kit - The migration fails in some cases [ED-4457]

### LINKED ISSUE #15477: Re-migrate globals not working as expected when upgrading from Elementor 2.x to 3.x
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

When upgrading from Elementor 2.x to 3.x, the re-migrate globals functionality does not seem to work correctly. Theme Styles, Fonts and other global settings are not being migrated correctly. Triggering the upgrade script again via _Tools -> Version Control -> Rerun Update Script_ causes the following error, resulting in the request failing with status 500 - server error:

`PHP Fatal error:  Uncaught Error: Cannot unset string offsets in /var/www/html/wp-content/plugins/elementor/core/upgrade/elementor-3-re-migrate-globals.php:77`

It seems like the upgrade script looks up the PostMeta value of `_elementor_page_settings` for the "Default Kit" post. When it does not exist, the lookup function returns a blank string value, but the rest of the upgrade logic expects the value to be an array. While processing the various style items that need to be migrated, the code attempts to unset specific entries from the array, which fails since calling `unset` on a string is invalid.

## Steps to reproduce

1. Set up a new site with Elementor 2.x (we used 2.19.14 with the Pro plugin at 2.10.3)
2. Set up a custom font, change the default page width, and apply a Theme Style background to all posts.
3. Ensure all other plugins are disabled, and that the default WordPress theme is active.
4. Upgrade the Elementor plugin to the latest stable version (currently 3.2.5 according to the WordPress plugins page)
5.  Go ahead and try to run the "Rerun Update Scripts" functionality from _Tools -> Version Control_

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>

```
        == Server Environment ==
	        Operating System: Linux
	        Software: nginx/1.17.5
	        MySQL version: Source distribution v8.0.20
	        PHP Version: 7.4.13
	        PHP Max Input Vars: 1000
	        PHP Max Post Size: 200M
	        GD Installed: Yes
	        ZIP Installed: Yes
	        Write Permissions: All right
	        Elementor Library: Connected
        
        == WordPress Environment ==
	        Version: 5.7.2
	        Site URL: <some internal company url>
	        Home URL: <some internal company url>
	        WP Multisite: Yes
	        Max Upload Size: 73 MB
	        Memory limit: 512M
	        Permalink Structure: /%postname%/
	        Language: en-US
	        Timezone: 0
	        Debug Mode: Inactive
        
        == Theme ==
	        Name: 
	        Version: 
	        Author: Kurtosys Systems
	        Child Theme: No
        
        == User ==
	        Role: administrator
	        WP Profile lang: en_US
	        User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:89.0) Gecko/20100101 Firefox/89.0
        
        == Network Plugins ==
	        Elementor
		        Version: 3.2.3
		        Author: Elementor.com
        
	        Elementor Pro
		        Version: 3.2.2
		        Author: Elementor.com
        
        
        == Elementor - Compatibility Tag ==
	        
        
        
        == Elementor Pro - Compatibility Tag ==
	        
        
        
        == Elements Usage ==
	        
	        header : 22
		        button : 17
		        heading : 17
		        html : 1
		        icon : 11
		        image : 2
		        nav-menu : 1
		        text-editor : 1
	        page : 211
		        blockquote : 2
		        button : 6
		        divider : 6
		        form : 1
		        heading : 13
		        icon : 2
		        icon-list : 1
		        image : 8
		        image-box : 3
		        text-editor : 12
		        video : 1
	        popup : 50
		        form : 1
		        heading : 1
	        section : 636
		        divider : 2
		        form : 27
		        heading : 4
		        image : 3
		        text-editor : 6
	        widget : 0
		        expandable-text : 1
        
        == Log ==

```
</details>

### PR DESCRIPTION
**Pull request by jp-vernooy**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Fix: Apply a default value for $kit_raw_settings in re-migrate-globals functionality

## Description

* See #15477 for an explanation of the problem being addressed
* Apply a default value for the $kit_raw_settings value before it gets used (in case the backing database entry wasn't found or didn't contain a valid value)

## Test instructions

* See #15477 for steps to reproduce the original issue. That issue should be resolved if this works correctly.

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [x] I have added unittests to verify the code works as intended
- [x] Docs have been added / updated (for bug fixes / features)

Fixes #15477

### CODE REVIEW

File: core/upgrade/elementor-3-re-migrate-globals.php [hunk 1]
```diff
@@ -71,6 +71,11 @@ private function run() {
 		$meta_key = SettingsPageManager::META_KEY;
 		$kit_raw_settings = $kit->get_meta( $meta_key );
 
+		// Apply a default value for $kit_raw_settings in case it's not defined.
+		if ( empty( $kit_raw_settings )) {
+			$kit_raw_settings = array();
```
Comment (matipojo) [inline core/upgrade/elementor-3-re-migrate-globals.php:76]:
```suggestion
			$kit_raw_settings = [];
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Comment by shilo-ey** [comment]
Rebased to 3.4.0

Please see the comment and fix. we will merge it after you will commit your changes.

Cheers.

File: core/upgrade/elementor-3-re-migrate-globals.php (hunk 1 above)
Comment (JasonWolmarans) [reply core/upgrade/elementor-3-re-migrate-globals.php:76]: @matipojo unrelated, but linked to this file

$active_kit_id  is undefined
Suggest adding in the run function? unless there is a reason for this

$active_kit_id = Plugin::$instance->kits_manager->get_active_id();

Thanks

Comment (matipojo) [reply core/upgrade/elementor-3-re-migrate-globals.php:76]: How can it be `undefined`? the get_active_kit will create it if it does not exist.

Comment (andy-culshaw) [reply core/upgrade/elementor-3-re-migrate-globals.php:76]: `$kit` is defined using `$kit = Plugin::$instance->kits_manager->get_active_kit();`, but `$active_kit_id` is not defined, but used here https://github.com/elementor/elementor/blob/3098d347d7af2db95299d36cc445165e9dcc1a6c/core/upgrade/elementor-3-re-migrate-globals.php#L102

Comment (matipojo) [reply core/upgrade/elementor-3-re-migrate-globals.php:76]: You are right, we will fix it. (@Aviad-Herman FYI)

Comment (JasonWolmarans) [reply core/upgrade/elementor-3-re-migrate-globals.php:76]: @matipojo looking at this [commit](https://github.com/elementor/elementor/commit/d7eeee2eadb5ba9819bc1f04239e7c013171e208)

The fix we spoke of above was in the code at some point, perhaps this got lost along the way?

Comment (jvernooy) [reply core/upgrade/elementor-3-re-migrate-globals.php]: Made the required changes.

**Review by matipojo — CHANGES_REQUESTED** [review]
Please fix the lint error: https://github.com/elementor/elementor/pull/15478/checks?check_run_id=3212684641#step:5:12

**Review by matipojo — APPROVED** [review]

**Pull request merged**